tests/validation/tests/v3_api/test_registry.py

```diff
@@ -115,8 +153,240 @@ def test_edit_registry_all_namespace():
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, token)
+    if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
+        with pytest.raises(ApiError) as e:
+            registry = \
+                create_registry_validate_workload(p_client, ns, allns=True)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'PermissionDenied'
+    else:
+        registry, workload = \
+            create_registry_validate_workload(p_client, ns, allns=True)
+        p_client.delete(workload)
+
+        # Create and validate workload in a new namespace
+        new_ns = create_ns(c_client, cluster, project)
+        workload = create_validate_workload(p_client, new_ns)
+        delete_registry(p_client, registry, ns)
+        p_client.delete(workload)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_delete_registry_single_namespace(role):
+    """
+    Runs only if RANCHER_TEST_RBAC is True in env. variable.
+    Creates a registry for single namespace, deploys a workload
+    and delete registry afterwards.
+    Validates the workload which has become invalid on registry deletion.
+    @param role: User role in rancher eg. project owner, project member etc
+    """
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, token)
+
+    if role in (CLUSTER_MEMBER, PROJECT_READ_ONLY):
+        with pytest.raises(ApiError) as e:
+            registry = create_registry_validate_workload(p_client, ns)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'PermissionDenied'
+    else:
+        registry, workload = create_registry_validate_workload(p_client, ns)
+        delete_registry(p_client, registry, ns)
+        p_client.delete(workload)
+
+        print("Verify workload cannot be created in the namespace after "
+              "registry deletion")
+        workload_invalid = \
+            create_validate_workload_with_invalid_registry(p_client, ns)
+        p_client.delete(workload_invalid)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_delete_registry_all_namespace(role):
+    """
+    Runs only if RANCHER_TEST_RBAC is True in env. variable.
+    Creates a registry scoped for all namespace, deploys a workload
+    and delete registry afterwards.
+    Validates the workload which has become invalid on registry deletion.
+    @param role: User role in rancher eg. project owner, project member etc
+    """
+    c_client = namespace["c_client"]
+    cluster = namespace["cluster"]
+    token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    p_client = get_project_client_for_token(project, token)
+    ns = rbac_get_namespace()
+    new_ns = create_ns(c_client, cluster, project)
```

Comment:
Create registry as cluster owner and the check for delete as different roles.

Reply:
Done